proxy/proxy/client.go

```diff
@@ -225,6 +228,15 @@ func (c *Client) Dial(instance string) (net.Conn, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	// Refresh cert 5 minutes before it expires.
+	timer := time.NewTimer(cfg.Certificates[0].Leaf.NotAfter.Sub(time.Now()) - refreshCertBuffer)
```

Comment:
You should definitely verify that the delay value passed to the time package is not negative since the cert info is from a remote server, or else the code will probably panic if that server ever serves a strange-looking cert.

Reply:
Done!